service/service.go

```diff
@@ -240,10 +242,23 @@ func New(config Config) (*Service, error) {
 		}
 	}
 
+	frameworkInitializer := func(ctx context.Context, obj interface{}) (context.Context, error) {
+		{
+			ch, err := configmapnamescontext.NewChannel(obj)
+			if err != nil {
+				return nil, microerror.Mask(err)
+			}
+			ctx = configmapnamescontext.NewContext(ctx, ch)
+		}
+
+		return ctx, nil
+	}
```

Comment:
I even started to like initializing the context after thinking of the logger context.